### PR TITLE
[TDV-1185, TDV-1158] Feature/integrate with session key server

### PR DESCRIPTION
**Pull request by Electr1Xx**

### CODE REVIEW

File: src/vuex/session.module.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  async [vuexTypes.UPDATE_SESSION] ({ commit, getters }) {
+    const sessionId = getters[vuexTypes.sessionId]
+    const { data } = await api.get(`/sessions/${sessionId}`)
+    return data
+  },
+}
+
+export const getters = {
+  [vuexTypes.sessionId]: state => state.sessionId,
+  [vuexTypes.encryptWalletSeed]: (state, getters) => state.encryptSecretSeed,
```
Comment (max-nes) [inline src/vuex/session.module.js]: `encryptedSeed`

File: src/vue/mixins/idle-handler.js [hunk 2]
```diff
@@ -36,5 +45,46 @@ export default {
...
+        this.setSessionKeeperInterval()
+      }
+      window.addEventListener('focus', () => {
+        this.keepSession()
+        this.setSessionKeeperInterval()
+      })
+      window.addEventListener('blur', () => {
+        clearInterval(this.sessionKeeperInterval)
+      })
+    },
```
Comment (max-nes) [inline src/vue/mixins/idle-handler.js]: something strange

File: src/vue/mixins/idle-handler.js [hunk 3]
```diff
@@ -36,5 +45,46 @@ export default {
...
+
+    async keepSession () {
+      try {
+        await this.updateSession()
+      } catch (error) {
```
Comment (max-nes) [inline src/vue/mixins/idle-handler.js]: check the error type

File: src/vuex/wallet.module.js [hunk 4]
```diff
@@ -32,17 +31,21 @@ export const actions = {
   async [vuexTypes.LOAD_WALLET] ({ commit }, { email, password }) {
     const wallet = await walletsManager.get(email, password)
     useWallet(wallet)
+    commit(vuexTypes.SET_SESSION, wallet)
     commit(vuexTypes.SET_WALLET, wallet)
   },
-  async [vuexTypes.STORE_WALLET] ({ commit }, wallet) {
+  async [vuexTypes.STORE_WALLET] ({ commit }, { wallet, session }) {
```
Comment (max-nes) [inline src/vuex/wallet.module.js]: should not accept `session`

File: src/vuex/idle-handler.module.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+const IDLE_TICKER_INTERVAL = 1000
+const THROTTLE_DELAY = 3000
+
+function getQueryParameters (queryParametr) {
+  // location.search return the querystring part of a URL
+  return location.search ? '&' + queryParametr : '?' + queryParametr
+}
+
+const state = {
+  logoutAt: null,
```
Comment (max-nes) [inline src/vuex/idle-handler.module.js]: null is prohibited default value

File: src/vue/App.vue [hunk 6]
```diff
@@ -152,6 +165,31 @@ export default {
...
+        location.reload()
+      }
+    },
+    watchChangesInLocalStorage () {
+      window.onstorage = (storage) => {
+        if (this[vuexTypes.isLoggedIn]) {
+          this.popState()
+          this.$router.push({
+            name: this.$route.name,
+          })
```
[the remark this answers is not in the document]
Comment (Electr1Xx) [reply src/vue/App.vue]: deleted

**Pull request merged**